app/code/Magento/Customer/view/frontend/templates/address/grid.phtml

```diff
@@ -63,7 +63,7 @@ $customerAddressView = $block->getData('customer_address');
         <button type="button" role="add-address" title="<?= $block->escapeHtmlAttr(__('Add New Address')) ?>" class="action primary add"><span><?= $block->escapeHtml(__('Add New Address')) ?></span></button>
     </div>
     <div class="secondary">
-        <a class="action back" href="<?= $block->escapeUrl($block->getBackUrl()) ?>"><span><?= $block->escapeHtml(__('Back')) ?></span></a>
+        <a class="action back" href="<?= $block->escapeUrl($block->getUrl('customer/account')) ?>"><span><?= $block->escapeHtml(__('Back')) ?></span></a>
```

Comment:
Why is it redirected to this page? The back button means real back, but not constantly going to the same page.